### DISCUSSION TITLE
Cut-off approach circles at edge of screen when screen scaling

### DISCUSSION THREAD

**Discussion by kukukhameleon** [ppy/osu#19985] [Ideas]
Would it be possible to add an option where the approach circle of a note can extend past where the screen normally ends when screen scaling?
![osu_2022-08-27_12-29-27](https://user-images.githubusercontent.com/112277083/187045390-6800dc16-5a30-4c2d-9280-f3e9fad94a06.jpg)
![osu_2022-08-27_12-29-27](https://user-images.githubusercontent.com/112277083/187045530-73f7e846-918e-4c1e-9b6c-78af4271e883.jpg)
I think it would make the gameplay feel a bit more clean especially with 100% background dim and without HUD

**Comment by peppy** [comment]
Use gameplay scale instead

**Reply by Feodor0090** [reply]
It will still happen:
![изображение](https://user-images.githubusercontent.com/53872073/187050818-bf754522-608e-4e54-b422-f1c6ea75edc4.png)

**Reply by peppy** [reply]
I believe this is quite intentional to not give a wider field of view by scaling out (less important for osu! ruleset, but more for the others).

I'm not sure we'll change this, but I guess if there was enough interest it could be switched off on rulesets which don't require this automatic masking.

**Comment by okolenmi** [comment]
As my discussion #20673 was marked as duplicate, I want to revive this thread. It's important thing. There must be some general solution to this. 
Personally, I think that there should be no restrictions on rendering objects. Maybe it would look like "extra possibilities" in some cases. I can understand this. But we will not be able to do something useful if we constantly rely on the old things.
Anyway this "extra possibilities" is still exist now. Mania as example:

![osu_2022-10-10_06-58-08](https://user-images.githubusercontent.com/55319814/194797474-edd8d888-c8eb-46a4-b5ef-412ff4545d69.jpg)

Looks funny as for me. But it's "stretching". No extra objects. Anyway it's possible to create the same picture with only Operation System monitor resolution settings (orientation and other things)...
For osu! there is no big effect. It's just cosmetic thing (playfield crop looks really ugly...)

In CTB it's fixed in this way:
![osu_2022-10-10_07-06-12](https://user-images.githubusercontent.com/55319814/194798143-dfba11b7-132e-415b-ba12-de6bed9e929b.jpg)

In Taiko it's "stretching" like in mania:
![osu_2022-10-10_07-07-07](https://user-images.githubusercontent.com/55319814/194798196-f13e8f91-1923-4511-89db-c3861b42732a.jpg)

Objects are closer to each other with another settings:
![osu_2022-10-10_07-15-53](https://user-images.githubusercontent.com/55319814/194798671-1ee44ae2-205e-4142-a8f3-ece381e582b2.jpg)

Just something "ultimate":
![osu_2022-10-10_08-02-25](https://user-images.githubusercontent.com/55319814/194801859-755a7c04-5d61-459e-add6-87f4b355ad75.jpg)
